### ISSUE TITLE
Accessibility: Status message not announced by screen readers [Product Single]

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52141]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The status message "67 in stock" is visually displayed when the size dropdown is updated, but screen readers do not announce it because it has not been programmatically defined as a status message.

![Image](https://github.com/user-attachments/assets/2468965d-1fe9-4f92-a9b0-33530dce75a4)

### Component or Page: 

Product Single

### Name affected element:

Price and in-stock inventory status message

### Relevant code snippets:

`<div class=""woocommerce-variation-availability""><p class=""stock in-stock"">67 in stock</p> </div>`

### Recommended fix:

The stock status should be announced by the screen reader. Make sure any status message displayed as visible text has the appropriate ARIA role:
 
 1. If the status message contains important, time-sensitive information that should be communicated to users immediately (potentially clearing the speech queue of previous updates), use role=""alert"", which has an implicit aria-live value of assertive.
 2. Otherwise, use a role with an implicit aria-live value of polite:
  a. Use role=""status"" for a simple status message that's not urgent enough to justify interrupting the current task.
  b. Use role=""log"" if new information is added to the status message in meaningful order, and old information might disappear (such as a chat log, game log, error log, or messaging history).
